### PR TITLE
feature/text tokens

### PR DESCRIPTION
**Pull request by sburba**

### CODE REVIEW

File: web/src/drag/DragStateTypes.ts [hunk 1]
```diff
@@ -1,25 +1,25 @@
 import Pos2d, { Bounds } from "../util/shape-math";
-import { Icon } from "../ui/icons";
+import { TokenContents } from "../types";
 
 export enum DraggableType {
-  Icon = "icon",
+  TokenSource = "token source",
```
[the remark this answers is not in the document]
Comment (sjohnson31) [reply web/src/drag/DragStateTypes.ts:5]: After reviewing the rest of this PR, I see that "source" is used in a lot of places, and perhaps this is too nitpicky to request a change.

File: api/src/game_components.py [hunk 2]
```diff
@@ -24,3 +37,12 @@ class Ping:
...
     y: int
+
+
+def content_id(contents: TokenContents) -> str:
+    if isinstance(contents, TextTokenContents):
+        return f"text-${contents.text}"
+    elif isinstance(contents, IconTokenContents):
+        return f"icon-${contents.icon_id}"
+    else:
+        raise ValueError(f"Unknown contents class type ${type(contents)}")
```
[the remark this answers is not in the document]
Comment (sburba) [reply api/src/game_components.py]: Good catch, fixed

File: web/src/drag/DragStateTypes.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sburba) [reply web/src/drag/DragStateTypes.ts]: Changed! I like token blueprint a lot

File: web/src/network/BoardStateApiClient.test.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    eventHandler = jest.fn();
+    client.setEventHandler(eventHandler);
+
+    client.connect("roomId");
+    await api.connected;
+
+    expect(eventHandler).toHaveBeenCalledWith({ type: EventType.Connected });
+  });
+
+  it("disconnects after a connection timeout", async () => {
```
Comment (sburba) [inline web/src/network/BoardStateApiClient.test.ts:84]: remove this

**Pull request merged**